src/core/diagram/qgsdiagram.h

```diff
@@ -39,6 +39,21 @@ class CORE_EXPORT QgsDiagram
 {
   public:
 
+#ifdef SIP_RUN
+    SIP_CONVERT_TO_SUBCLASS_CODE
+    if ( sipCpp->diagramName() == QStringLiteral( "Pie" ) )
```

Comment:
Nice; that sip conversion could probably make its way into 3.10 (minus the stacked bit)

Reply:
Yep, definitely. (that said, it's actually not that vital given that the subclasses don't really expose anything useful)